pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -573,10 +573,11 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         if self.dirpath is not None:
             return  # short circuit
 
+        # TODO: Remove weights_save_path logic here in v1.8
         if trainer.logger is not None:
-            if trainer.weights_save_path != trainer.default_root_dir:
+            if trainer._weights_save_path_internal != trainer.default_root_dir:
```

Comment:
```suggestion
            if trainer._weights_save_path != trainer.default_root_dir:
```
if you really mean it as internal, lets make is private or protected

Reply:
I'm confused - it is private already?

Reply:
oh I think I know what you mean. I don't want to just change `weights_save_path` to `_weights_save_path` because we already have `_weights_save_path` which does something else:
https://github.com/PyTorchLightning/pytorch-lightning/blob/dc4c3171fc0618fb463a846ccbde2e558902303b/pytorch_lightning/trainer/trainer.py#L2207-L2215